### ISSUE TITLE
Make Assets Pallet's Privileged Roles `Option`

### ISSUE THREAD

**Issue by joepetrowski** [paritytech/polkadot-sdk#91]
Change the [privileged asset roles](https://github.com/paritytech/substrate/blob/73c4f94ce0cb41e35bd7fbc7095590b98a351dbb/frame/assets/src/types.rs#L34L41) from `AccountId` to `Option<AccountId>`. When the `owner` is set to `None`, only governance would be able to restore the role.

For discussion, if `owner` is `Some`, and they want to set a lower rank role from `None` to `Some`, should that require governance?

This would look similar to the [`PoolRoles` struct](https://github.com/paritytech/substrate/blob/73c4f94ce0cb41e35bd7fbc7095590b98a351dbb/frame/nomination-pools/src/lib.rs#L528) in Nomination Pools. 

_Originally posted by @joepetrowski in https://github.com/paritytech/substrate/pull/12310#discussion_r982367220_

**Comment by thiolliere** [comment]
Removing ownership priviledge is a common practice in some coin like meme coin for instance.
* PINK token recently revoke ownership by using a trick: move ownership to pure proxy and kill the proxy. 
  This is a bit like inscription, prooving an ownership has been revoked is not light-client friendly
* solana TOKEN 2022 contract has a function revoke freeze and mint. (and metadata?)

Actually I am only interested in a feature to revoke every priviledge, I feel like revoking only mint, or only freezer, or only admin is not really useful.

2 implementation:
* fine grained: make Owner, Freezer, Issuer, Admin an option and add some call to revoke them individually
  it is a breaking change, needs migration, frontend will have to update a bit.
* Easy implementation: make a new asset status: `LiveAndNotOwned`.
  doesn't break the storage, frontend only needs to handle this new status.

**Comment by joepetrowski** [comment]
Good point about the easy implementation, `None` roles for anyone but the owner is kind of meaningless. I would maybe go for something more like `LiveAndLocked` as a status.

**Comment by Captain1653** [comment]
Hey! 

I was looking into this issue and wanted to pick it up, but noticed a blocker regarding recent changes in the codebase.

With PR #7579, the fields in `AssetDetails` were intentionally made `pub` (previously they were `pub(super)`). Changing these fields to `Option<AccountId>` now would introduce a major breaking change to the public API contract, alongside the required storage migrations.

Given that this issue has been open for a while and the pallet's API visibility has evolved since then,  should this issue be closed as obsolete?
